Detectors/Upgrades/ITS3/macros/test/buildMatBudLUT.C

```diff
@@ -0,0 +1,411 @@
+    auto newBuff = new char[mbrC->getFlatBufferSize()];
+    auto newObj = new char[sizeof(*mbrC)];
+    auto oldBuff = mbrC->releaseInternalBuffer();
+    mbrC->setFutureBufferAddress(newBuff);
+    memcpy(newObj, mbrC, sizeof(*mbrC));
+    memcpy(newBuff, oldBuff, mbrC->getFlatBufferSize());
+    o2::base::MatLayerCylSet* mbrF = (o2::base::MatLayerCylSet*)newObj; // !!! this is the object to use
+    //<<< end of the lines needed to copy the object
+
+    delete mbrC;      // delete cloned object and its buffer which are now in a
+    delete[] oldBuff; // dirty state
+    mbrC = nullptr;
+
+    // check created object
+    gSystem->RedirectOutput("matbudFuture.txt", "w");
+    mbrF->print(true);
+    gSystem->RedirectOutput(nullptr);
+
+    auto diff = gSystem->Exec("diff matbudFuture.txt matbudActual.txt");
+    if (diff) {
+      LOG(error) << "Difference between cloned at created at /FutureBuffer/ LUTs";
+      return false;
+    }
+  }
+  return true;
+}
+
+//_______________________________________________________________________
+void configLayers()
+{
+  const int NSect = 18;
+
+  const float kToler = 1e-3; //
+  float drStep = 0.f, zSpanH = 0.f, zBin = 0.f, rphiBin = 0.f, phiBin = 0.f;
+
+  o2::itsmft::ChipMappingITS mp;
+  int nStave = 0;
+  //                           rMin    rMax   zHalf
+  lrData.emplace_back(LrData(0.0f, 1.65f, 30.f));
+
+  // beam pipe
+  lrData.emplace_back(LrData(lrData.back().rMax, 1.8f, 30.f));
+  //===================================================================================
+  // ITS3 Inner Barrel
+  drStep = 0.2;
+  zSpanH = 20.;
+  rphiBin = 0.2; // 0.1
+  zBin = 0.5;
+  float drStepChip = 0.053;
+  float drStepChip4Layer = 0.075;
+
+  // Layer 1
+  lrData.emplace_back(LrData(1.8f, 1.8f + drStepChip, zSpanH, zBin, rphiBin));
+  // Air
+  do {
+    lrData.emplace_back(LrData(lrData.back().rMax, lrData.back().rMax + drStep, zSpanH, zBin, rphiBin));
+  } while (lrData.back().rMax < 2.4f - drStep + kToler);
+  lrData.emplace_back(LrData(lrData.back().rMax, 2.4f, zSpanH, zBin, rphiBin));
+
+  // Layer 2
+  lrData.emplace_back(LrData(lrData.back().rMax, lrData.back().rMax + drStepChip, zSpanH, zBin, rphiBin));
+  // Air
+  do {
+    lrData.emplace_back(LrData(lrData.back().rMax, lrData.back().rMax + drStep, zSpanH, zBin, rphiBin));
+  } while (lrData.back().rMax < 3.0f - drStep + kToler);
+  lrData.emplace_back(LrData(lrData.back().rMax, 3.0f, zSpanH, zBin, rphiBin));
+
+  // Layer 3
+  lrData.emplace_back(LrData(lrData.back().rMax, lrData.back().rMax + drStepChip, zSpanH, zBin, rphiBin));
+  // Air
+  do {
+    lrData.emplace_back(LrData(lrData.back().rMax, lrData.back().rMax + drStep, zSpanH, zBin, rphiBin));
+  } while (lrData.back().rMax < 6.0f - drStep + kToler);
+  lrData.emplace_back(LrData(lrData.back().rMax, 6.0f, zSpanH, zBin, rphiBin));
+
+  // Layer 4
+  lrData.emplace_back(LrData(lrData.back().rMax, lrData.back().rMax + drStepChip4Layer, zSpanH, zBin, rphiBin));
```

Comment:
Maybe we can have an argument that allows us to turn on/off the addition of the 4th layer to have the possibility to build the LUT both for the 3-layer and the 4-layer versions